### DISCUSSION TITLE
Plugin for laser cutting and engraving with bCNC

### DISCUSSION THREAD

**Discussion by Git-Bruno** [vlachoudis/bCNC#1864] [Show and tell]
Hi,
I have written a plugin to prepare selected blocks for laser cutting and engraving. 
In a nut shell it has 3 functions:
1) Cleanup F,S and Z move commands in the selected blocks.
2) Add Fxxx and Sxxxx gcode to selected blocks.
3) Repeat blocks to achieve multiple passes, optionnaly with a Z down motion at each pass.

Tabs generated by bCNC are supported by the plugin. More documentation can be found in file LaserCut.py.

I have tested the plugin to prepare gcode for imported dxf files or collected gcode, originally created for milling, and I have been using the modified gcode on my 3018 CNC without any problem.

I am not sure about the best way to share this code or get it added to bCNC so I attach it here in a gz file. Comments and suggestions are welcome.
Bruno

[LaserCut.gz](https://github.com/vlachoudis/bCNC/files/13588523/LaserCut.gz)

**Comment by Harvie** [comment]
> I am not sure about the best way to share this code or get it added to bCNC 

Usualy on github the process is following:

1.) Use FORK button to create your own fork of bCNC
2.) Use git tools to clone your fork, commit and push your changes to it
3.) Create pull request ("PR" for short) that requests maintainers of original vlachoudis/bCNC to pull from your fork (eg. Git-Bruno/bCNC)
4.) Monitor merge process in https://github.com/vlachoudis/bCNC/pulls

**Comment by Git-Bruno** [comment]
Thanks for the hints Harvie. I believe I completed the steps OK to create the PR although I don't see it in the list of open PR yet so I guess it is now waiting for somebody to pull it and review.
